File: Daily Test/Python/Finding Missing Digit - Sum 28.8.2025.py
n=int(input())
a=input().split()
s=int(input())
for i in range(n):
    if "*" in a[i]:
        for d in range(10):
            x=int(a[i].replace("*",str(d)))
            if sum(int(y.replace("*","0")) if "*" in y else int(y) for y in a)-int(a[i].replace("*","0"))+x==s:
                a[i]=str(x);break
print(*a)

File: Daily Test/Java/Finding Missing Digit - Sum 28.8.2025.java
import java.util.*;

class Main {
    public static void main(String[] args){
        Scanner sc=new Scanner(System.in);
        int n=sc.nextInt();
        String[] a=new String[n];
        for(int i=0;i<n;i++) a[i]=sc.next();
        int s=sc.nextInt();
        for(int i=0;i<n;i++){
            if(a[i].contains("*")){
                for(int d=0;d<=9;d++){
                    String t=a[i].replace("*",String.valueOf(d));
                    int x=Integer.parseInt(t),sum=0;
                    for(int j=0;j<n;j++){
                        if(j==i) sum+=x;
                        else sum+=Integer.parseInt(a[j].replace("*","0"));
                    }
                    if(sum==s){a[i]=t;break;}
                }
            }
        }
        for(String x:a) System.out.print(x+" ");
    }
}